src/Modules/UngradedModules.jl

```diff
@@ -0,0 +1,2689 @@
+export presentation
+
+abstract type ModuleFP{T} end
+
+const CRing = Union{MPolyRing, MPolyQuo{<:Oscar.MPolyElem}, MPolyRing_dec, MPolyQuo{<:Oscar.MPolyElem_dec}}
+const CRingElem = Union{MPolyElem, MPolyQuoElem{<:Oscar.MPolyElem}, MPolyElem_dec, MPolyQuoElem{<:Oscar.MPolyElem_dec}}
+#TODO: "fix" to allow QuoElem s as well...
+# this requires
+#  re-typeing of FreeModule
+#  typing of BiModArray
+# ... and all the rest.
+# parametrization has to be by elem_type(coeff_ring) and not, like currently, the bottom coeff ring
+# Also: qring is a Singular native. So it needs to be added to the ring creation
+
+abstract type ModuleFPHom end
+abstract type ModuleMap{T1, T2} <: Map{T1, T2, Hecke.HeckeMap, ModuleFPHom} end
+
+
+# For all module types, including free modules we store canonical in and outgoing morphisms 
+# which were generated by the function creating the module
+# otherwise a free module consists out of a rank and a ring
+# one can also provide names for the generators
+mutable struct FreeMod{T} <: ModuleFP{T}
+  R::CRing
+  n::Int
+  S::Array{Symbol, 1}
+
+  ingoing_morphisms::Array{<:ModuleMap,1}
+  outgoing_morphisms::Array{<:ModuleMap,1}
+
+  AbstractAlgebra.@declare_other
+
+  function FreeMod(n,b::CRing,c)
+    r = new{elem_type(b)}()
+    r.n = n
+    r.R = b
+    r.S = c
+
+    r.ingoing_morphisms = Array{ModuleMap,1}()
+    r.outgoing_morphisms = Array{ModuleMap,1}()
+
+    return r
+  end
```

Comment:
As discussed: This is not actually a constructor, as its name is not equal to the typename, which is `FreeMod{T}`! This is a nasty pitfall in Julia, which can lead to confusing issues later on. Please read more o n this at https://docs.julialang.org/en/v1/manual/constructors/#Parametric-Constructors (and perhaps study that complete section carefully)

```suggestion
  function FreeMod{T}(n,b::CRing,c) where {T}
    return new(R, n, c, Vector{ModuleMap}(), Vector{ModuleMap}())
  end
```
I would also suggest to sync the order of arguments to the order of how the elements of the struct are defined (`R` first or `n` first?) and also the names (why `c` vs `s` and `b` vs. `R`)

Also, if you can, specify the types of the arguments